libexec/bats-core/bats

```diff
@@ -363,13 +464,23 @@ for filename in "${arguments[@]}"; do
 
   if [[ -d "$filename" ]]; then
     shopt -s nullglob
-    if [[ "$recursive" -eq 1 ]]; then
+    if [[ "$recursive" == "0" ]]; then
+      build_find_command_string "find_command"
+
+      # shellcheck disable=SC2154 
       while IFS= read -r -d $'\0' file; do
-        filenames+=("$file")
-      done < <(find -L "$filename" -type f -name "*.${BATS_FILE_EXTENSION:-bats}" -print0 | sort -z)
+        filenames+=("$file")   
+        # performance boost through piping by /dev/fd/63 instead of
+        # creating a temporary file for find's results.
+        # => pipe [<(..)] and redirect [<] '$find_command' to standard input.
+        # for `read` using c-style string termination '\0' [-print0]
+        # eval is a built-in command, no performance issue's expected
+      done < <(eval "$find_command" -print0 | sort -z)
```

Comment:
Can we have some performance data that drove this decision? While I am all for performance, I am even more for readable code. This is hardly on bats' hot path.

Reply:
i am totally on your side with clean code (readability).

that's why i thought it would be useful to briefly document how this loop works. 
i have not changed this existing main construct. only the find is now dynamically built, which led to the use of eval. if there are better variants, i am very open and happy to learn. thanks.